hathor/transaction/scripts/opcode.py

```diff
@@ -157,7 +157,7 @@ def op_pushdata1(position: int, full_data: bytes, stack: Stack) -> int:
     return new_pos
 
 
-def op_dup(stack: Stack, log: list[str], extras: ScriptExtras) -> None:
```

Comment:
Why have you removed the log in some opcodes?

Reply:
Because they were not being used. Most opcodes don't use all 3 parameters (`stack`, `log`, and `extras`), and since I'm going to add a fourth one (`context`), I made this change so each opcode can have a different signature, eliminating unused parameters.

Reply:
Should we create a context with all of them inside it? I mean, `Context` would carry the stack, log, extras, and whatever other stuff we might need in the future. So we don't have to change the method signatures anymore.

Reply:
Sure, done in https://github.com/HathorNetwork/hathor-core/pull/851/commits/f35564bb1f1696c6fdb87d1066646d8a60b0a3c1

Reply:
Great! Thanks!